### PR TITLE
Technology create wizard

### PR DESCRIPTION
**Pull request by GingerBoy14**

### CODE REVIEW

File: src/app/domains/Technology/routes/TechnologyCreate/TechnologyCreate.layout.js [hunk 1]
```diff
@@ -1,32 +1,369 @@
...
+
+    for (const subLevel of Object.values(
+      selectedPreset.levelIds[targetOption.value]
+    )) {
+      const subLevelSnapshot = await firestore
+        .collection(COLLECTIONS.SUB_LEVELS)
+        .doc(subLevel)
+        .get()
+
+      const subLevelName = subLevelSnapshot.data().name
```
Comment (VladBogach) [inline src/app/domains/Technology/routes/TechnologyCreate/TechnologyCreate.layout.js:202]: #doc_existing_check

File: src/app/domains/Technology/routes/TechnologyCreate/TechnologyCreate.layout.js [hunk 2]
```diff
@@ -1,32 +1,369 @@
...
+
+      targetOption.children.push({ label: subLevelName, value: subLevel })
+    }
+    targetOption.loading = false
+
+    setLevelTree([...levelTree])
+  }
+
+  const onLevelSelect = (value) => {
```
Comment (VladBogach) [inline src/app/domains/Technology/routes/TechnologyCreate/TechnologyCreate.layout.js:211]: #fix_naming value => ... (not clear)

File: src/app/domains/Technology/routes/TechnologyCreate/TechnologyCreate.layout.js [hunk 3]
```diff
@@ -1,32 +1,369 @@
...
+  // [USE_EFFECTS]
+  useEffect(() => {
+    historyState?.formData && mainForm.setFieldsValue(historyState.formData)
+  }, [mainForm, historyState])
+
+  useEffect(() => {
+    //TODO need to set initLoading
+    const initTechnology = async () => {
+      try {
+        const technology = await firestore
```
Comment (VladBogach) [inline src/app/domains/Technology/routes/TechnologyCreate/TechnologyCreate.layout.js:245]: #getCollectionId

File: src/app/domains/Technology/routes/TechnologyCreate/TechnologyCreate.layout.js [hunk 4]
```diff
@@ -1,32 +1,369 @@
...
+                          </Space>
+                        </Box>
+                      </Col>
+                      <Col span={24}>
+                        <Space
+                          direction="vertical"
+                          align="center"
+                          style={{ width: '100%' }}>
+                          <Text>Required technologies: </Text>
+                          <Form.List
```
Comment (VladBogach) [inline src/app/domains/Technology/routes/TechnologyCreate/TechnologyCreate.layout.js:327]: #structure
Move list to separated component

File: src/app/domains/Todo/components/lists/TodoSimpleList/TodoSimpleList.template.js [hunk 5]
```diff
@@ -60,37 +72,45 @@ const TodoSimpleList = (props) => {
...
+                    <Form.Item
+                      style={{ flex: 1 }}
+                      name="todo"
+                      noStyle
+                      hasFeedback={editLoading}
+                      validateStatus="validating">
                       <Input.TextArea
+                        disabled={editLoading}
                         onPressEnter={(e) => {
                           e.preventDefault()
```
Comment (VladBogach) [inline src/app/domains/Todo/components/lists/TodoSimpleList/TodoSimpleList.template.js:101]: #add_coments why preventing default event

File: src/app/domains/Todo/routes/TodoCreate/TodoCreate.layout.js [hunk 6]
```diff
@@ -20,36 +22,105 @@ const TodoCreate = () => {
...
-    value && setTodos([...todos, value])
-    setEditTodo('')
+  const onSubmit = async (value) => {
+    if (value) {
+      setTodoAddLoading(true)
+
+      try {
+        const collectionRef = firestore.collection(COLLECTIONS.TODOS)
+
+        const todoRef = await collectionRef.add({})
```
Comment (VladBogach) [inline src/app/domains/Todo/routes/TodoCreate/TodoCreate.layout.js:48]: #getCollectionId

File: src/app/domains/Todo/routes/TodoCreate/TodoCreate.layout.js [hunk 7]
```diff
@@ -20,36 +22,105 @@ const TodoCreate = () => {
...
+      delete historyState?.todoIds[levelId][subLevelId]
+      // Check if there was last item is this level delete level object
+      if (!Object.values(historyState?.todoIds[levelId]).length) {
+        delete historyState?.todoIds[levelId]
+      }
+    }
+    history.push(historyState.prevLocation, historyState)
   }
-  // -----------------------------
+  // ----------------------------------
```
Comment (VladBogach) [inline src/app/domains/Todo/routes/TodoCreate/TodoCreate.layout.js:110]: nice changes (please remove comment)

**Review by olenashevchuk — APPROVED** [review]

**Pull request merged**